### PR TITLE
Slow synchronization to 3.2.2cn

### PR DESCRIPTION
**Pull request by jzy-chitong56**
this hope help synchronization to 3.2.2cn

when you merge completed or close , I will submit the next one when I have time

----------------------------------------------------
**new** 
**ITEM_EXPANSION JOB and ITEM_EXPANSION_CHECK JOB**
refer to ANCIENT_EXPANSION , help AI use item passive EXPANSION

**ITEM REPEAT CHECK JOB**
Check for items that cannot be superposition or have poor superposition effects , transfer to other heroes

----------------------------------------------------
**change**
CopyGroup use new api

----------------------------------------------------
**fix**
- AMAI VS AI cannot work
- ROC use PickMeleeHero choose hero no set neutral hero
- ver_neutral_hero_number Initial value is 0 ，call InitGlobalSettings will set new value  , then ohter AI can use PickMeleeHero choose hero , and now ROC still is 0
- GOBLIN_TINKER abnormal skill learning
- ENDURANCE_AURA now is AOr2 , C should is Cairne

----------------------------------------------------
by the way ， 2.6.2cn code consistent with A  ， so 2.6.2cn I won't submit the  Slow version , when you synchronization 3.2.2cn , then 2.6.2cn review workload will minimal

### CODE REVIEW

File: TFT/HeroAbilities.txt
Comment (SMUnlimited) [inline TFT/HeroAbilities.txt:1]: These changes are incorrect. These are special abilities for campaign heroes.

Comment (jzy-chitong56) [reply TFT/HeroAbilities.txt:1]: It's not true, but the four character encoding corresponding to this item is duplicated，So I can only suspect it was written incorrectly

Comment (SMUnlimited) [reply TFT/HeroAbilities.txt:1]: AOr2 is only used by Cairn in the world editor and has 4 upgrade slots instead of just 3.

Comment (jzy-chitong56) [reply TFT/HeroAbilities.txt:1]: So how to deal with this repetition
Directly delete or not process？？

ENDURANCE_AURA	AOae
ENDURANE_AURA	AOae

**Comment by SMUnlimited** [comment]
Merged via feature/122

**Pull request closed without merging**